cesm/driver/ensemble_driver.F90

```diff
@@ -187,40 +225,67 @@ subroutine SetModelServices(ensemble_driver, rc)
     call NUOPC_CompAttributeGet(ensemble_driver, name="ninst", value=cvalue, rc=rc)
     if (chkerr(rc,__LINE__,u_FILE_u)) return
     read(cvalue,*) number_of_members
+    
+    call NUOPC_CompAttributeGet(ensemble_driver, name="pio_async_iotasks", value=cvalue, rc=rc)
+    if (chkerr(rc,__LINE__,u_FILE_u)) return
+    read(cvalue,*) pio_async_iotasks
 
-    call ESMF_VMGet(vm, localPet=localPet, PetCount=PetCount, rc=rc)
+    call NUOPC_CompAttributeGet(ensemble_driver, name="pio_async_iostride", value=cvalue, rc=rc)
     if (chkerr(rc,__LINE__,u_FILE_u)) return
+    read(cvalue,*) pio_async_iostride
 
-    ntasks_per_member = PetCount/number_of_members
-    if(ntasks_per_member*number_of_members .ne. PetCount) then
+    call ESMF_VMGet(vm, localPet=localPet, PetCount=PetCount, rc=rc)
+    if (chkerr(rc,__LINE__,u_FILE_u)) return
+    
+    ntasks_per_member = PetCount/number_of_members - pio_async_iotasks
+    if(ntasks_per_member*number_of_members .ne. (PetCount - pio_async_iotasks)) then
        write (msgstr,'(a,i5,a,i3,a,i3,a)') &
-            "PetCount (",PetCount,") must be evenly divisable by number of members (",number_of_members,")"
+            "PetCount - Async IOtasks (",PetCount-pio_async_iotasks,") must be evenly divisable by number of members (",number_of_members,")"
        call ESMF_LogSetError(ESMF_RC_ARG_BAD, msg=msgstr, line=__LINE__, file=__FILE__, rcToReturn=rc)
        return
     endif
 
+    if(pio_async_iotasks > 0 .and. .not. asyncIO_available) then
```

Comment:
Write a message to the driver log file saying that you are using async io
It would be helpful to have this in log output